src/mysky.ts

```diff
@@ -176,27 +214,23 @@ export class MySky {
     return signature;
   }
 
-  async userID(_opts?: CustomUserIDOptions): Promise<string> {
-    // Get the seed.
+  async checkPermission(path: string, category: PermCategory, permType: PermType): Promise<void> {
+    // Check for the permissions provider.
 
-    const seed = checkStoredSeed();
-    if (!seed) {
-      throw new Error("User seed not found");
+    if (!permissionsProvider) {
+      throw new Error("Permissions provider not loaded");
     }
 
-    // Get the public key.
-
-    const { publicKey } = genKeyPairFromSeed(seed);
-    return publicKey;
+    const referrerDomain = await this.client.extractDomain(referrer);
+    const perm = new Permission(referrerDomain, path, category, permType);
+    log(`Checking permission: ${JSON.stringify(perm)}`);
+    const connection = await permissionsProvider;
+    const resp: CheckPermissionsResponse = await connection.remoteHandle().call("checkPermissions", [perm], dev);
+    if (resp.failedPermissions.length > 0) {
+      const readablePerm = readablePermission(perm);
+      throw new Error(`Permission was not granted: ${readablePerm}`);
+    }
```

Comment:
`permissionProvider`, `referrer`, and `dev` are all globals which makes reading this rather confusing. Would it be possible to either:
1. pass them as parameters, so at least the magic is condensed to a single spot
2. group them into a common object (passed as parameter or not) which describes their nature (a good name goes a long way)
3. name them in an indicative way, so the reader immediately knows where these come from

My personal preference would be 2 but any of the others work as well. If that's for some reason impractical then I won't block the PR for it but... it would be way more readable if those are somehow denoted as special.

Reply:
I did my best to make this less confusing but I'm not sure what to do for `permissionsProvider`. It can be modified by the top-level event listener so it's a non-constant global. Maybe could be part of a larger refactor in the future.